[14.0][ADD]  shopfloor_full_location_reservation: Full location reservation for shopfloor

mt-software-de opened the pull request.

mt-software-de committed:
[ADD] shopfloor_full_location_reservation: Full location reservation for shopfloor

mt-software-de commented:
cc @jbaudoux

mt-software-de commented:
Tests are missing

jbaudoux commented on shopfloor_full_location_reservation/services/zone_picking.py:
```diff
@@ -0,0 +1,41 @@
+    _inherit = "shopfloor.zone.picking"
+
+    def _handle_full_location_package_reservation(self, package):
+        return package and self.work.menu.full_location_reservation
```
It is not a full location reservation, it is a full package reservation.

If you had scanned a product not part of a package, it should trigger a full product reservation. Same for lot.

jbaudoux commented on shopfloor_full_location_reservation/models/shopfloor_menu.py:
```diff
@@ -0,0 +1,27 @@
+    full_location_reservation_is_possible = fields.Boolean(
+        compute="_compute_full_location_reservation_is_possible"
+    )
+    full_location_reservation = fields.Boolean(
```
Shouldn't this be replaced by `full_source_reservation` ?

jbaudoux commented:
This was initially developed for the `location content transfer` scenario but it doesn't apply properly to other scenarios. I think this should be slightly adapted to become more generic.

cc @rousseldenis

rousseldenis committed:
[FIX] shopfloor_full_location_reservation: Don't do the full reservation on move but on move line

As a stock move can point to several move lines on several stock locations, don't do
the full reservation process on move but on lines.

mt-software-de commented:
@jbaudoux @rousseldenis This is now used quite some time in production. How should we move forward with this?
I cherry-picked your fix @rousseldenis https://github.com/OCA/wms/pull/798/commits/b4cffd415e5fe67bddd6d62d32ba0e4cbeb8a1f2
IMO this PR and the v16 PR would be ready to merge.

jbaudoux commented:
/ocabot merge nobump
